src/JSONSettingsRESTAPI/Endpoints/SettingsSetValueEndpoint.cpp

```diff
@@ -0,0 +1,90 @@
+#include "stdafx.h"
+#include "SettingsSetValueEndpoint.h"
+
+#include "Helpers/ReplyBuilderHelper.h"
+#include "Helpers/SettingCurrentValueHelper.h"
+#include "Helpers/SettingLookupHelper.h"
+#include "Helpers/SettingSerializationHelper.h"
+#include "Helpers/SettingValueParsingHelper.h"
+
+#include "JSONSettings/ISettingsService.h"
+#include "JSONSettings/SettingDefinitionMgr.h"
+
+#include "JSONAdapterInterface/IJSONAdapter.h"
+#include "JSONAdapterInterface/IJSONDocument.h"
+#include "JSONAdapterInterface/IJSONValue.h"
+
+#include "RESTAPICore/Endpoint/EndpointRequestData.h"
+#include "WebServerAdapterInterface/Model/Reply.h"
+
+
+namespace systelab { namespace setting { namespace rest_api {
+
+	SettingsSetValueEndpoint::SettingsSetValueEndpoint(const SettingsFile& settingsFile,
+													   std::unique_ptr<ISettingsService> settingsService,
+													   const systelab::json::IJSONAdapter& jsonAdapter)
+		:m_settingsFile(settingsFile)
+		,m_settingsService(std::move(settingsService))
+		,m_jsonAdapter(jsonAdapter)
+	{
+	}
+
+	SettingsSetValueEndpoint::~SettingsSetValueEndpoint() = default;
+
+	std::unique_ptr<systelab::web_server::Reply> SettingsSetValueEndpoint::execute(const systelab::rest_api_core::EndpointRequestData& endpointRequestData)
+	{
+		try
+		{
+			if (!endpointRequestData.getParameters().hasNumericParameter("id"))
+			{
```

Comment:
OK, before looking the test, I see there are other types,  I didn't realize that I only review some changes, not the entire source. Sorry for this.

Reply:
This parameter refers to the "id" contained on the endpoint route and used to identify the setting. It is a number autogenerated by the library every time you register a setting and its unique within the settings file. Thus, as this will be always a value, this parameter needs to be numeric. Handling of types for the values of the setting comes afterwards.